### PR TITLE
Update custom context class option to be compatible with new GraphQLContext class

### PR DESCRIPTION
**Pull request by oryan-block**
Also minimize use of deprecated APIs. 

## Checklist
- [x] Pull requests follows the [contribution guide](https://github.com/graphql-java-kickstart/graphql-java-tools/wiki/Contribution-guide)
- [x] New or modified functionality is covered by tests

## Description
Following https://github.com/graphql-java/graphql-java/pull/2368 the custom context class option is deprecated in GraphQL Java, this PR does the same for Tools by using the "official" `GraphQLContext` class ~~and deprecating the `contextClass` option~~.

The `contextClass` option is still available as a value in the new context object with the context class as the key:
Clients that want to keep using this option will have to adjust accordingly:

```
var executionInput = ExecutionInput.newExecutionInput()
            .query(query)
            .variables(variables)
            .graphQLContext(Map.of(CustomContext.class, context));
```

`environment.<GraphQLContext>getContext();` -> `environment.getGraphQLContext().<CustomContext>get(CustomContext.class);`

### CODE REVIEW

File: src/main/kotlin/graphql/kickstart/tools/resolver/MethodFieldResolver.kt [hunk 1]
```diff
@@ -94,7 +96,11 @@ internal class MethodFieldResolver(
         if (this.additionalLastArgument) {
             when (this.method.parameterTypes.last()) {
                 null -> throw ResolverError("Expected at least one argument but got none, this is most likely a bug with graphql-java-tools")
-                options.contextClass -> args.add { environment -> environment.getContext() }
+                options.contextClass -> args.add { environment ->
+                    environment.graphQlContext[options.contextClass]
+                        ?: environment.getContext() // TODO: remove deprecated use in next major release
```
Comment (oryan-block) [inline src/main/kotlin/graphql/kickstart/tools/resolver/MethodFieldResolver.kt:101]: No breaking change as of yet. Next version will be though.

File: src/main/kotlin/graphql/kickstart/tools/resolver/MethodFieldResolver.kt [hunk 2]
```diff
@@ -94,7 +96,11 @@ internal class MethodFieldResolver(
         if (this.additionalLastArgument) {
             when (this.method.parameterTypes.last()) {
                 null -> throw ResolverError("Expected at least one argument but got none, this is most likely a bug with graphql-java-tools")
-                options.contextClass -> args.add { environment -> environment.getContext() }
+                options.contextClass -> args.add { environment ->
+                    environment.graphQlContext[options.contextClass]
```
Comment (vojtapol) [inline src/main/kotlin/graphql/kickstart/tools/resolver/MethodFieldResolver.kt:100]: Let's print a warning whenever the right part of the `?:` operator gets hit.

This is how I would word it roughly:

```
Generic context class has been deprecated by graphql-java. To continue using a custom context class as the last parameter in resolver methods please insert it into the new GraphQLContext class when building the ExecutionInput.
```

File: src/test/kotlin/graphql/kickstart/tools/MethodFieldResolverDataFetcherTest.kt [hunk 3]
```diff
@@ -176,20 +177,19 @@ class MethodFieldResolverDataFetcherTest {
...
+        assertEquals(resolver.get(createEnvironment(DataClass(), context = context)), true)
     }
 
     @Test
     fun `data fetcher passes context if method has extra argument and context is specified`() {
-        val context = ContextClass()
-        val options = SchemaParserOptions.newOptions().contextClass(ContextClass::class).build()
-        val resolver = createFetcher("active", options = options, resolver = object : GraphQLResolver<DataClass> {
-            fun isActive(dataClass: DataClass, ctx: ContextClass): Boolean {
+        val context = GraphQLContext.newContext().build()
```
Comment (vojtapol) [inline src/test/kotlin/graphql/kickstart/tools/MethodFieldResolverDataFetcherTest.kt:190]: This test needs to be changed back to test out that custom context classes still work. Or maybe split it in two tests if we want both.

**Review by vojtapol — CHANGES_REQUESTED** [review]

File: src/test/kotlin/graphql/kickstart/tools/MethodFieldResolverDataFetcherTest.kt (hunk 3 above)
Comment (oryan-block) [reply src/test/kotlin/graphql/kickstart/tools/MethodFieldResolverDataFetcherTest.kt:190]: Added a new test to reflect that.

**Review by vojtapol — APPROVED** [review]

**Pull request merged**